webapp/src/contexts/ConferenceContext/methods/changeEffect.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  type VideoProcessor,
+  createCanvasTransform,
+  createSegmenter,
+  createVideoProcessor,
+  createVideoTrackProcessor,
+  createVideoTrackProcessorWithFallback,
+  type ProcessVideoTrack
+} from '@pexip/media-processor'
+import { ConferenceActionType, type ConferenceAction } from '../ConferenceAction'
+import { type Effect } from 'src/types/Effect'
+import { type ConferenceState } from '../ConferenceState'
+
+interface NavigatorUABrandVersion {
+  brand: string
+  version: string
+}
+
+interface NavigatorUAData {
+  brands: NavigatorUABrandVersion[]
+  mobile: boolean
+  platform: string
+}
+
+let processor: VideoProcessor | null = null
+
+export const changeEffect = async (
+  localVideoStream: MediaStream,
+  effect: Effect,
+  state: ConferenceState,
+  dispatch: React.Dispatch<ConferenceAction>
+): Promise<MediaStream> => {
+  processor?.close()
+
+  // await processor?.destroy()
+
+  processor = await getVideoProcessor(effect)
+
+  await processor.open()
+
+  const processedVideoStream = await processor.process(localVideoStream)
+
+  let newStream: MediaStream
+  if (state.localAudioStream != null) {
+    newStream = new MediaStream([...processedVideoStream.getTracks(), ...state.localAudioStream.getTracks()])
+  } else {
+    newStream = processedVideoStream
+  }
+
+  state.client?.setStream(newStream)
+
+  dispatch({
+    type: ConferenceActionType.UpdateLocalStream,
+    body: {
+      processedVideoStream
+    }
+  })
+
+  dispatch({
+    type: ConferenceActionType.ChangeEffect,
+    body: {
+      effect
+    }
+  })
+
+  return processedVideoStream
+}
+
+let baseUrl = document.currentScript?.getAttribute('src')
+
+if (baseUrl != null) {
+  baseUrl = baseUrl.substring(0, baseUrl.lastIndexOf('/')) + '/public/'
+  baseUrl = baseUrl.replace('/static/', '/')
+} else {
+  baseUrl = window.location.origin + '/static/plugins/com.pexip.pexip-video-connect/public/'
+}
+
```

Comment:
should you use `URL` obj to create validate all of this urls instead of doing any replacing?

Reply:
The idea is change the URL from something like this:

```
https://api.pexip.tech/mattermost/static/plugins/com.pexip.pexip-video-connect/com.pexip.pexip-video-connect_48ffc651d5b0eb6a_bundle.js
```

To:

```
https://api.pexip.tech/mattermost/plugins/com.pexip.pexip-video-connect/public/
```

I don't know a better way to do it.

BTW, I will add a new commit with some small style corrections.

Reply:
```js
new URL('/mattermost/plugins/com.pexip.pexip-video-connect/public/', 'https://api.pexip.tech/mattermost/static/plugins/com.pexip.pexip-video-connect/com.pexip.pexip-video-connect_48ffc651d5b0eb6a_bundle.js').toString()
```
 Have you try smth like this?

Reply:
But I still don't see the advantages and neither know of to apply it with `new URL`:

- It works if the plugin is served from a subpath (e.g. /mattermost in that case).
- It's independent of the plugin_id.

The `else` is mainly to pass the tests.